internal/client/segments.go

```diff
@@ -0,0 +1,75 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/terraform-provider-grafana-adaptive-metrics/internal/model"
+)
+
+const (
+	segmentsEndpoint = "/aggregations/rules/segments"
+)
+
+func (c *Client) CreateSegment(s model.Segment) (model.Segment, error) {
+	c.segmentMutex.Lock()
+	defer c.segmentMutex.Unlock()
+
+	body, err := json.Marshal(s)
+	if err != nil {
+		return model.Segment{}, err
+	}
```

Comment:
could the marshaling be done outside of the mutex lock to shorten the lock time?